### PR TITLE
Refactor: use kubenetes client instead of kuberay apiserver

### PR DESCRIPTION
**Pull request by akihikokuroda**
### Summary
This PR refactor the gateway ray cluster creation and deletion 

### Details and comments
This replace the kuberay apiserver use with the kubernetes dynamic client.

### CODE REVIEW

**Comment by psschwei** [comment]
https://github.com/Qiskit-Extensions/quantum-serverless/pull/642 will fix the failing client test

File: gateway/api/ray.py
Comment (IceKhan13) [reply gateway/api/ray.py:77]: and as I said before I'm really liking this approach :) much more flexible then hitting kuberay limitations

Comment (akihikokuroda) [reply gateway/api/ray.py:77]: @IceKhan13  Jinja templates looks good.  I also thinking about putting the template in a ConfigMap. So we don't need to touch the container image to change the configuration and probably we can use the helm template to render some part.  I'll do these improvement in a follow up PR.  Thanks!

File: gateway/api/ray.py
Comment (IceKhan13) [inline gateway/api/ray.py:72]: can we include those settings here and we are good to merge :) Thank you, Aki!

File: gateway/api/ray.py
Comment (psschwei) [reply gateway/api/ray.py:79]: > putting the template in a ConfigMap
+100

File: gateway/api/ray.py
Comment (psschwei) [inline gateway/api/ray.py:13]: out of curiosity, is there a reason to prefer the openshift dynamic client to the vanilla kubernetes one?

File: infrastructure/helm/quantumserverless/values.yaml
Comment (psschwei) [inline infrastructure/helm/quantumserverless/values.yaml:234]: is there anything still using the api server? if not, may as well rip off the bandaid and cut all this stuff out now :smile:

Comment (akihikokuroda) [reply infrastructure/helm/quantumserverless/values.yaml:234]: No, nothing is using the api server.  It can be taken off along with the raycluster.

**Comment by akihikokuroda** [comment]
I'll run `helm dependency update` and push Chart.lock.

**Comment by IceKhan13** [comment]
Aki, I will test it today (or latest tomorrow), I'm buried in meetings today

**Comment by pacomf** [comment]
@akihikokuroda @psschwei if we are getting 500 errors with the current code... and tests are not failing... can we do something to detect it from tests?

**Comment by pacomf** [comment]
> @akihikokuroda @psschwei if we are getting 500 errors with the current code... and tests are not failing... can we do something to detect it from tests?

maybe the error is related with the @caleb-johnson error that it is appearing after add a new test for the docker environment: https://github.com/Qiskit-Extensions/quantum-serverless/actions/runs/5195660406/jobs/9368597527?pr=648, so i dont know if this test certify the @psschwei error as well, so we can have a test to jump into the error in our CI

**Comment by akihikokuroda** [comment]
@psschwei I can not see the issue here.  What are in the log of the gateway-scheduler pod?  Thanks!

File: gateway/api/ray.py
Comment (akihikokuroda) [reply gateway/api/ray.py:74]: I can make it in the values.yaml file

Comment (psschwei) [reply gateway/api/ray.py:74]: that would require reploying the helm chart for any updates though...

either way, let's hold off on that change until we merge this PR

File: gateway/api/ray.py
Comment (akihikokuroda) [reply gateway/api/ray.py:13]: The openshift one just adds "apply" function.  We don't use it so I'll change to the kubernetes one in the follow up PR.  Thanks for pointing it out.

**Review by IceKhan13 — APPROVED** [review]
Checked! Looks like everything works as expected!

In following PRs we can remove helm kuberay and cluster charts and only leave ray operator. This will simplify deployment configs.

Great work, Aki!

**Review by IceKhan13 — APPROVED** [review]

**Comment by akihikokuroda** [comment]
Thanks!

**Comment by akihikokuroda** [comment]
I'm using Rancher Desktop.  I don't do anything except port forwarding Jupyter notebook service.

**Comment by IceKhan13** [comment]
yes, I also do port forwarding

**Comment by psschwei** [comment]
So no jupyter ... how are you testing programs?
(I'm running notebooks in jupyter... I wonder if that's why I'm seeing issues)

**Comment by akihikokuroda** [comment]
The gateway-scheduler should create the instance.  Do you see any exceptions in the gateway-scheduler pod?

**Comment by IceKhan13** [comment]
> So no jupyter ... how are you testing programs?

I have local jupyter with local programs and connecting to gateway straight away

**Comment by psschwei** [comment]
> Is the raycluster CRD instance created?

yes, it's there

> Do you see any exceptions in the gateway-scheduler pod?

yes, I'm still seeing this one:

```
Defaulted container "gateway-scheduler" out of: gateway-scheduler, waitpostresql (init)
Using selector: EpollSelector
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UndefinedTable: relation "api_job" does not exist
LINE 1: ...d", "api_job"."ray_job_id", "api_job"."logs" FROM "api_job" ...
                                                             ^
```

> I have local jupyter with local programs 

Local notebooks fail too (though I'm guessing the scheduler error above has something to do with it).

**Comment by IceKhan13** [comment]
oh, yes we have a call tomorrow for exactly this :)

**Comment by caleb-johnson** [comment]
I'm working on #652 and seeing gateway failures in CI but not locally running same commands

**Pull request merged**
